orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v2schema/model/V2PipelineTemplate.java

```diff
@@ -0,0 +1,307 @@
+    }
+
+    public void setNullable(boolean nullable) {
+      this.nullable = nullable;
+    }
+
+    @Override
+    public boolean isMerge() {
+      return merge;
+    }
+
+    public void setMerge(boolean merge) {
+      this.merge = merge;
+    }
+
+    @Override
+    public boolean isRemove() {
+      return remove;
+    }
+
+    public void setRemove(boolean remove) {
+      this.remove = remove;
+    }
+
+    @Override
+    public Variable merge(Variable overlay) {
+      Variable v;
+      try {
+        v = (Variable) this.clone();
+      } catch (CloneNotSupportedException e) {
+        throw new RuntimeException("Could not clone Variable", e);
+      }
+      if (overlay.group != null) { v.group = overlay.group; }
+      if (overlay.description != null) { v.description = overlay.description; }
+      if (overlay.type != null) { v.type = overlay.type; }
+      if (overlay.defaultValue != null) { v.defaultValue = overlay.defaultValue; }
+      if (overlay.example != null) { v.example = overlay.example; }
+      return v;
+    }
+
+    @Override
+    protected Object clone() throws CloneNotSupportedException {
+      return super.clone();
+    }
+  }
+
+  public static class Configuration {
+    private Map<String, Object> concurrentExecutions;
+    private List<NamedHashMap> triggers;
+    private List<NamedHashMap> parameters;
+    private List<NamedHashMap> notifications;
+    private List<HashMap<String, Object>> expectedArtifacts;
+
+    public Map<String, Object> getConcurrentExecutions() {
+      return Optional.ofNullable(concurrentExecutions).orElse(new HashMap<>());
+    }
+
+    public void setConcurrentExecutions(Map<String, Object> concurrentExecutions) {
+      this.concurrentExecutions = concurrentExecutions;
+    }
+
+    public List<NamedHashMap> getTriggers() {
+      return triggers;
+    }
+
+    public void setTriggers(List<NamedHashMap> triggers) {
+      this.triggers = triggers;
+    }
+
+    public List<NamedHashMap> getParameters() {
+      return parameters;
+    }
+
+    public void setParameters(List<NamedHashMap> parameters) {
+      this.parameters = parameters;
+    }
+
+    public List<NamedHashMap> getNotifications() {
+      return notifications;
+    }
+
+    public void setNotifications(List<NamedHashMap> notifications) {
+      this.notifications = notifications;
+    }
+
+    public List<HashMap<String, Object>> getExpectedArtifacts() {
+      return expectedArtifacts;
+    }
+
+    public void setExpectedArtifacts(List<HashMap<String, Object>> expectedArtifacts) {
+      this.expectedArtifacts = expectedArtifacts;
+    }
+  }
+
+  @Override
+  @JsonIgnore
+  public String getSchemaVersion() {
+    return schema;
+  }
+
+  public String getSchema() {
+    return schema;
+  }
+
+  public void setSchema(String schema) {
+    this.schema = schema;
+  }
+
+  public String getId() {
+    return id;
+  }
+
+  public void setId(String id) {
+    this.id = id;
+  }
+
+  public Metadata getMetadata() {
```

Comment:
All of this boilerplate should be replaced by lombok

Reply:
Done.